coalib/core/Core.py

```diff
@@ -327,34 +149,192 @@ def instantiate_and_track(prev_bear_type, next_bear_type):
     return dependency_tracker, bears
 
 
+class Session:
+    """
+    Maintains a session for a coala execution. For each session, there are set
+    of bears to run along with a callback function, which is called when
+    results are available.
+
+    If BearA requires BearB as dependencies, then, on running BearA,
```

Comment:
you maybe want to prepend some general statement before the example, like

Dependencies of bears (provided via ``bear.BEAR_DEPS``) are automatically handled.
If BearA requires ...